developブランチでアプリ開発

osugitaro opened the pull request:
developブランチにてアプリを作成しました。
以下の仕様は満たされていると思います。
 ・トップ画面(エリア検索、フリーワードで検索できること)

 ・ユーザ登録機能(プロフィール情報として、アイコン・自己紹介・名前を登録できること。一つのアカウントで貸す機能、借りる機能どちらもできること)

 ・貸したい物件を登録できること(住所、価格、フリーテキスト、画像を登録できる)

 ・借りたい物件を予約できること

 ・予約履歴を参照できること

実装するにあたり、迷った点がいくつかあります。それは
・スネークケースとキャメルケースはどちらが良いのか？
・cssは複数のページでまたがっても、classでまとめた方が良いのか？
です。

レビューの程、よろしくお願いいたします。

Commit:
developブランチでアプリ開発

Commit:
rails課題1で指摘されたところを修正

osugitaro commented:
前回の課題「スケジュール管理アプリ」で指摘された以下の点を修正いたしました。
・reservation.rbのメソッドをself.start_date < self.end_date へと簡略化。

・pathの指定をprefixにする。

・行末の改行コードを入れる。

・全体のインデントを半角スペース2つ分にする。

MasahiroMatsuno commented:
![スクリーンショット 2022-12-05 10 40 24](https://user-images.githubusercontent.com/64773002/205530887-b84caed4-5a7c-441c-93c7-68de14dc72c6.png)
部屋の詳細に遷移する際にエラーが発生しているようです。user登録から一通り動作確認をお願いいたします🙏

Commit:
部屋の詳細画面に飛ぶ際に、ユーザー画像がない場合は初期画像を載せるように修正

osugitaro commented:
部屋の詳細画面に飛ぶ際にユーザー画像がないとエラーが出てしまっていました。

そのため、ユーザー画像がないときには初期画像を表示するように以下のファイルを修正いたしました。
app/views/reservations/new.html.erb

MasahiroMatsuno commented:
![スクリーンショット 2022-12-05 16 58 06](https://user-images.githubusercontent.com/64773002/205583947-24cb145f-4246-4022-a9c3-d7978feb77ee.png)
バリデーションがかかって正常な値でも予約ができないようです。
提出前にユーザー登録から一通りの動作確認をお願いいたします🙏

Commit:
・バリデーション(start_end_check)の不等号が逆になっていたため、修正
・ユーザーが消えたら、予約済みと作成したルームが消えるようにdependent: :destroyを追加
・eメールとパスワードのバリデーションをcreateの時にのみ発生するように修正

osugitaro commented:
以下の3つを修正しました。

・Reservationのバリデーションメソッドstart_end_checkの不等号が逆になっており、エラーが発生していました。そのため不等号を逆に修正いたしました。

・ユーザーのレコードを削除した際に、ユーザーが作成した部屋が残る、予約済みのデータが残るということが発生していました。そのためuser.rbにてdependent: :destroyを追加しました。

・ユーザー情報を更新するときに、emailとpasswordのバリデーションpresence: trueを入れるとエラーが発生していました。そのためpresence: trueをcreateの時にだけ発生するようにonメソッドを加えました。

MasahiroMatsuno commented on Gemfile:
```diff
@@ -52,5 +52,28 @@ group :test do
+
+gem 'jquery-rails'
+
+gem 'mini_magick'
```
複数のgem活用できていていいですね👍

MasahiroMatsuno commented on app/assets/stylesheets/common.scss:
```diff
@@ -0,0 +1,211 @@
+}
+
+
+
```
改行不要です！

MasahiroMatsuno commented on app/assets/stylesheets/rooms.scss:
```diff
@@ -0,0 +1,118 @@
+        max-width: 25%;
+    }
+    
+}
```
行末は改行コードをいれるのが慣習になっています。このファイルだけではないのですが、挿入されていないので警告がでています。
（㊀マーク）エディタのプラグインによって自動挿入してくれる機能もありますので調べてみてください。
https://bottoms-programming.com/archives/indent-tab-space-and-new-line.html

MasahiroMatsuno commented on app/controllers/reservations_controller.rb:
```diff
@@ -0,0 +1,47 @@
+    def reserved
+      @user = current_user
+      @reservations = @user.reservations
+    end
```
基本的にインデントは半角スペース2つ分を一段、改行で間を開けるときは1行で十分ということを意識して修正してみてください。

MasahiroMatsuno commented on app/controllers/rooms_controller.rb:
```diff
@@ -0,0 +1,30 @@
+    end
+
+    def create
+      @room = Room.new(params.require(:room).permit(:title, :introduction, :price, :address, :image))
```
private配下に置くようにしましょう！
https://qiita.com/ozackiee/items/f100fd51f4839b3fdca8

```
def create
  @post = Post.new(post_params)
  #...
end

def update
  @post = Post.find(params[:id])
  if @post.update(post_params)
  #...
end

def destroy
  # ...
end

private

def post_params
  params.require(:post).permit(:title, :start_date, :end_date, :all_day, :memo)
end
```

MasahiroMatsuno commented on app/models/reservation.rb:
```diff
@@ -0,0 +1,26 @@
+
+
+    def cannot_day_past
+    if start_date.past?
```
インデント下げましょう

MasahiroMatsuno reviewed (commented):
修正できていますね！LGTMです！
カリキュラムでの学びやカリキュラム外のプラスαの学びなども積極的に活かして実装できていたため素晴らしかったです。

以下アドバイスです。
(プルリク内のコメントも確認しておいてください！)
・コミットメッセージを細かく残す
https://qiita.com/itosho/items/9565c6ad2ffc24c09364
・Pull Requestのdescriptionに実装した機能の概要や実装の際に迷った点やレビュアーに特に見てほしい点などを書くようにしましょう。
・画像などレビューに関係ないファイルは差分にあげないようにしましょう
・テストについてもRspecを使っていきますので、
Everyday Rails等の書籍や、本書を訳された伊藤さんのqiitaの記事等を参考に学習を進めてみてください👍
https://leanpub.com/everydayrailsrspec-jp

以上、宿泊予約サービスアプリはLGTMです。
最終課題も気を抜かずがんばっていきましょう。
お疲れ様でした！
